test/WebJobs.Script.Tests/CSharpEndToEndTests.cs

```diff
@@ -170,6 +173,43 @@ public CSharpEndToEndTests(TestFixture fixture) : base(fixture)
             Assert.Equal("secondary type value", request.Properties["DependencyOutput"]);
         }
 
+        [Fact]
+        public async Task Scenario_RandGuidBinding_GeneratesRandomIDs()
+        {
+            var container = Fixture.BlobClient.GetContainerReference("scenarios-output");
+            if (container.Exists())
+            {
+                foreach (CloudBlockBlob blob in container.ListBlobs())
+                {
+                    await blob.DeleteAsync();
+                }
+            }
+
+            // Call 3 times - expect 3 separate output blobs
+            for (int i = 0; i < 3; i++)
+            {
+                ScenarioInput input = new ScenarioInput
+                {
+                    Scenario = "randGuid",
+                    Container = "scenarios-output",
+                    Value = i.ToString()
+                };
+                Dictionary<string, object> arguments = new Dictionary<string, object>
+                {
+                    { "input", JsonConvert.SerializeObject(input) }
+                };
+                await Fixture.Host.CallAsync("Scenarios", arguments);
+            }
+
+            var blobs = container.ListBlobs().Cast<CloudBlockBlob>().ToArray();
+            Assert.Equal(3, blobs.Length);
```

Comment:
Before the rand-guid changes I made to core SDK, since C# is early bound, only a single blob would be written (overwritten twice), since the %rand-guid% was resolved once at index/bind time.